### PR TITLE
New Python template without pipenv

### LINKED ISSUE #324: New Python template for pip without pipenv
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

pipenv is really slow and sometimes it hangs for a really long time.

Any objection to removing it? I think leaving the Python env setup to the user is fine - and just recommending Pyenv + virtualenv. Perhaps writing out a simple requirements.txt file instead of a Pipfile in new templates and docs.

Example where it hangs for me - perhaps because I'm using a private PyPi instance that makes it worse, but that works fine with plain pip.

```
Installing dependencies from Pipfile.lock (8d4353)…
```

### References

https://github.com/pypa/pipenv/issues/3827

### PR DESCRIPTION
**Pull request by cmclaughlin**

### CODE REVIEW

**Comment by cmclaughlin** [comment]
This seems to work well...  

```
$ CDKTF_DIST=~/repos/terraform-cdk/dist/ cdktf init --template python-pip --local --project-name=test --project-description=test
Note: By supplying '--local' option you have chosen local storage mode for storing the state of your stack.
This means that your Terraform state file will be stored locally on disk in a file 'terraform.tfstate' in the root of your project.
Processing /Users/cmclaughlin-local/repos/terraform-cdk/dist/python/cdktf-0.0.0-py3-none-any.whl
Requirement already satisfied: jsii<2.0.0,>=1.11.0 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from cdktf==0.0.0->-r requirements.txt (line 1)) (1.11.0)
Requirement already satisfied: constructs<4.0.0,>=3.0.0 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from cdktf==0.0.0->-r requirements.txt (line 1)) (3.0.4)
Requirement already satisfied: publication>=0.0.3 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from cdktf==0.0.0->-r requirements.txt (line 1)) (0.0.3)
Requirement already satisfied: python-dateutil in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from jsii<2.0.0,>=1.11.0->cdktf==0.0.0->-r requirements.txt (line 1)) (2.8.1)
Requirement already satisfied: attrs~=19.3.0 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from jsii<2.0.0,>=1.11.0->cdktf==0.0.0->-r requirements.txt (line 1)) (19.3.0)
Requirement already satisfied: cattrs~=1.0.0 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from jsii<2.0.0,>=1.11.0->cdktf==0.0.0->-r requirements.txt (line 1)) (1.0.0)
Requirement already satisfied: typing-extensions~=3.7.4 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from jsii<2.0.0,>=1.11.0->cdktf==0.0.0->-r requirements.txt (line 1)) (3.7.4.2)
Requirement already satisfied: six>=1.5 in /Users/cmclaughlin-local/.pyenv/versions/3.8.2/envs/cdktf/lib/python3.8/site-packages (from python-dateutil->jsii<2.0.0,>=1.11.0->cdktf==0.0.0->-r requirements.txt (line 1)) (1.15.0)
Installing collected packages: cdktf
  Found existing installation: cdktf 0.0.15
    Uninstalling cdktf-0.0.15:
      Successfully uninstalled cdktf-0.0.15
Successfully installed cdktf-0.0.0
WARNING: You are using pip version 19.2.3, however version 20.2.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Generated python constructs in the output directory: imports
========================================================================================================

  Your cdktf Python project is ready!

  cat help                Prints this message

  Compile:
    python3 ./main.py     Compile and run the python code.

  Synthesize:
    cdktf synth           Synthesize Terraform resources to cdktf.out/

  Diff:
    cdktf diff            Perform a diff (terraform plan) for the given stack

  Deploy:
    cdktf deploy          Deploy the given stack

  Destroy:
    cdktf destroy         Destroy the given stack

========================================================================================================

$ cat requirements.txt
/Users/cmclaughlin-local/repos/terraform-cdk/dist/python/cdktf-0.0.0-py3-none-any.whl
```

**Comment by skorfmann** [comment]
There was a similar build error in #329 - Is this branch up to date with master?

**Comment by cmclaughlin** [comment]
Thanks for the tip! I rebased and the tests are passing now.

File: docs/getting-started/python.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 - [Terraform](https://www.terraform.io/downloads.html) >= v0.12
 - [Node.js](https://nodejs.org) >= v12.16
 - [Python](https://www.python.org/downloads/) >= v3.7
-- [Pipenv](https://pipenv.pypa.io/en/latest/install/#installing-pipenv/)
+- Optional: [Pipenv](https://pipenv.pypa.io/en/latest/install/#installing-pipenv/) (see below)
```
Comment (skorfmann) [inline docs/getting-started/python.md:8]: Let's move this into the `pipenv` section below.

**Review by skorfmann — COMMENTED** [review]
Looks very good, thanks! 

Besides from the comment, an integration similar to [this one](https://github.com/hashicorp/terraform-cdk/tree/76da576abd9223861615144e5fe278c483850ea7/test/test-python-app) would be nice. However, if you're not comfortable adding this, I'm happy to merge and add it later.

**Comment by skorfmann** [comment]
> Besides from the comment, an integration similar to [this one](https://github.com/hashicorp/terraform-cdk/tree/76da576abd9223861615144e5fe278c483850ea7/test/test-python-app) would be nice. However, if you're not comfortable adding this, I'm happy to merge and add it later.

You know what - forget about this. We'll do https://github.com/hashicorp/terraform-cdk/issues/338 instead. So - just the little documentation change for now.

**Review by skorfmann — APPROVED** [review]

**Pull request merged**